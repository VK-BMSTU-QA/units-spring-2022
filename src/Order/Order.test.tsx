import React from 'react';
import {OrderComponent} from './Order';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
jest.mock('../utils/getDate');
import {getDate} from '../utils/getDate';

configure({ adapter: new Adapter() });


describe('Order.tsx', () => {
	let wrapper;
	it('Casual order', () => {
		getDate.mockReturnValue('1 января, чт, 1970 год');
		const order: Order = {
			id: 1,
			date: 17032022,
			shop: 'MegaShop',
			items: ['Бургер', 'Кола']
		};

		const key = 0;

		wrapper = shallow(<OrderComponent key = {key} order = {order}/>);
		expect(wrapper).toMatchSnapshot();
	});

	it('no items', ()=>{
		getDate.mockReturnValue('1 января, чт, 1970 год');
		const order: Order = {
			id: 1,
			date: 17032022,
			shop: 'MegaShop',
			items: {}
		};
	
		const key = 0;
	
		wrapper = shallow(<OrderComponent key = {key} order = {order}/>);
	
		expect(wrapper).toMatchSnapshot();
	});

	it('no date', ()=>{
		getDate.mockReturnValue(null);
		const order: Order = {
			id: 1,
			shop: 'MegaShop',
			items: ['Бургер', 'Кола']
		};
	
		const key = 0;
	
		wrapper = shallow(<OrderComponent key = {key} order = {order}/>);

		expect(wrapper).toMatchSnapshot();
	});

	it('no order', ()=>{
		getDate.mockReturnValue(null);
		const order: Order = {};
	
		const key = 0;
	
		wrapper = shallow(<OrderComponent key = {key} order = {order}/>);

		expect(wrapper).toMatchSnapshot();
	});
});
