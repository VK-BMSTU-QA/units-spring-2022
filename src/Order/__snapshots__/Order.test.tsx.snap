// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order.tsx render order with items 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      Alihandro Express
    </span>
    <span
      id="date"
    >
      17 марта, чт, 2022 год
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div
      className="Order-item"
      key="0"
    >
      Утиный пластмасса для показ новый год
    </div>
    <div
      className="Order-item"
      key="1"
    >
      Курица из нержавеющей стали, утка, гусь, голубь, питьевой фонтан
    </div>
    <div
      className="Order-item"
      key="2"
    >
      Новый стиль один розница яйцо для упаковки форма латекс
    </div>
  </div>
</div>
`;

exports[`Order.tsx render order without items 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      Сбереги Мега Маркер
    </span>
    <span
      id="date"
    >
      17 марта, чт, 2022 год
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div>
      Заказ пуст...
    </div>
  </div>
</div>
`;
